-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e45b4d/20782/summary.html
**COMMIT**: a64ad0c9a724949318c14b41ef8ec141fb9d20c4
**CMSSW**: CMSSW_12_0_X_2021-11-25-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/36254/20782/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-11-25-1100/src/DQM/GEM/plugins/GEMDQMHarvester.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-11-25-1100/src/DQM/GEM/plugins/GEMDigiSource.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-11-25-1100/src/DQM/GEM/plugins/GEMRecHitSource.cc
>> Building  edm plugin tmp/slc7_amd64_gcc900/src/DQM/GEM/plugins/DQMGEMPlugins/libDQMGEMPlugins.so
/cvmfs/cms-ib.cern.ch/nweek-02708/slc7_amd64_gcc900/external/gcc/9.3.0/bin/../lib/gcc/x86_64-unknown-linux-gnu/9.3.0/../../../../x86_64-unknown-linux-gnu/bin/ld: tmp/slc7_amd64_gcc900/src/DQM/GEM/plugins/DQMGEMPlugins/GEMDigiSource.cc.o:(.data.rel.ro._ZTV13GEMDigiSource[_ZTV13GEMDigiSource]+0xc0): undefined reference to `GEMDigiSource::ProcessWithMEMap2(GEMDQMBase::BookingHelper&, std::tuple<int, int>)'
collect2: error: ld returned 1 exit status
gmake: *** [tmp/slc7_amd64_gcc900/src/DQM/GEM/plugins/DQMGEMPlugins/libDQMGEMPlugins.so] Error 1
Leaving library rule at src/DQM/GEM/plugins
Entering library rule at DQM/GEM
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-11-25-1100/src/DQM/GEM/src/GEMDQMBase.cc
>> Building  shared library tmp/slc7_amd64_gcc900/src/DQM/GEM/src/DQMGEM/libDQMGEM.so
</pre>

--- a/DQM/GEM/interface/GEMDigiSource.h
+++ b/DQM/GEM/interface/GEMDigiSource.h
@@ -40,6 +40,7 @@ protected:
 
 private:
   int ProcessWithMEMap2WithEta(BookingHelper& bh, ME3IdsKey key) override;
+  int ProcessWithMEMap2(BookingHelper& bh, ME2IdsKey key) override;
   int ProcessWithMEMap3(BookingHelper& bh, ME3IdsKey key) override;
   int ProcessWithMEMap3WithChamber(BookingHelper& bh, ME4IdsKey key) override;
 
@@ -53,7 +54,7 @@ private:
   MEMap3Inf mapDigiOcc_phi_;
   MEMap3Inf mapTotalDigiPerEvtLayer_;
   MEMap3Inf mapTotalDigiPerEvtIEta_;
-  MEMap3Inf mapBX_iEta_;
+  MEMap2Inf mapBX_;
 
   MEMap4Inf mapDigiOccPerCh_;
 

--- a/DQM/GEM/interface/GEMDigiSource.h
+++ b/DQM/GEM/interface/GEMDigiSource.h
@@ -40,7 +40,6 @@ protected:
 
 private:
   int ProcessWithMEMap2WithEta(BookingHelper& bh, ME3IdsKey key) override;
-  int ProcessWithMEMap2(BookingHelper& bh, ME2IdsKey key) override;
   int ProcessWithMEMap3(BookingHelper& bh, ME3IdsKey key) override;
   int ProcessWithMEMap3WithChamber(BookingHelper& bh, ME4IdsKey key) override;
 

--- a/DQM/GEM/plugins/GEMDigiSource.cc
+++ b/DQM/GEM/plugins/GEMDigiSource.cc
@@ -58,7 +58,7 @@ void GEMDigiSource::bookHistograms(DQMStore::IBooker& ibooker, edm::Run const&, 
                                       "Events");
   mapTotalDigiPerEvtIEta_.SetNoUnderOverflowBin();
 
-  mapBX_iEta_ = MEMap3Inf(this, "bx", "Digi Bunch Crossing", 21, nBXMin_ - 0.5, nBXMax_ + 0.5, "Bunch crossing");
+  mapBX_ = MEMap2Inf(this, "bx", "Digi Bunch Crossing", 21, nBXMin_ - 0.5, nBXMax_ + 0.5, "Bunch crossing");
 
   mapDigiOccPerCh_ = MEMap4Inf(this, "occ", "Digi Occupancy", 1, -0.5, 1.5, 1, 0.5, 1.5, "Digi", "iEta");
 
@@ -80,8 +80,13 @@ void GEMDigiSource::bookHistograms(DQMStore::IBooker& ibooker, edm::Run const&, 
   }
 }
 
+int GEMDigiSource::ProcessWithMEMap2(BookingHelper& bh, ME2IdsKey key) {
+  mapBX_.bookND(bh, key);
+
+  return 0;
+}
+
 int GEMDigiSource::ProcessWithMEMap2WithEta(BookingHelper& bh, ME3IdsKey key) {
-  mapBX_iEta_.bookND(bh, key);
   mapTotalDigiPerEvtIEta_.bookND(bh, key);
 
   return 0;
@@ -183,7 +188,7 @@ void GEMDigiSource::analyze(edm::Event const& event, edm::EventSetup const& even
         // Filling of bx
         Int_t nBX = std::min(std::max((Int_t)d->bx(), nBXMin_), nBXMax_);  // For under/overflow
         if (bTagVFAT.find(nIdxVFAT) == bTagVFAT.end()) {
-          mapBX_iEta_.Fill(key3IEta, nBX);
+          mapBX_.Fill(key2, nBX);
         }
 
         // Occupancy on a chamber

